<launch>
	<node pkg="mantis_kinematics" type="mantis_kinematics_node" name="mantis_kinematics" clear_params="true" output="screen">

		<!-- Input Topics -->
		<param name="topic_marker_detected" value="detected_markers" type="str"/>
		<param name="topic_state_shoulder" value="/shoulder_controller/state" type="str"/>
		<param name="topic_state_elbow" value="/elbow_controller/state" type="str"/>
		<param name="topic_state_wrist" value="/wrist_controller/state" type="str"/>

		<!-- Output Topics -->
		<param name="topic_command_shoulder" value="/shoulder_controller/command" type="str"/>
		<param name="topic_command_elbow" value="/elbow_controller/command" type="str"/>
		<param name="topic_command_wrist" value="/wrist_controller/command" type="str"/>

		<param name="input_boundary" value="/home/pryre/catkin_ws/src/mantis_uav/mantis_config/config/bounds.txt" type="str"/>

		<!-- Frame Configuration -->
		<param name="mav_frame" value="fcu" type="str"/>
		<param name="camera_frame" value="usb_cam" type="str"/>
		<!-- Camera Transformation [X,Y,Z, w, x, y, z] -->
		<rosparam param="static_camera_pose">[0.125, 0.0, 0.0, 0.393, -0.588, 0.588, -0.393]</rosparam>
		<!-- Arm Transformation [w, x, y, z] -->
		<!-- This should be the rotation of mav_frame to the pivot of the mounting joint-->
		<rosparam param="static_arm_rotation">[0.707, 0.0, 0.707, 0.0]</rosparam>
		<param name="arm_length_mount" value="0.053" type="double"/>
		<param name="arm_length_link1" value="0.192" type="double"/>
		<param name="arm_length_link2" value="0.189" type="double"/>
	</node>
</launch>
